scipy/special/cephes/stirling.c

```diff
@@ -111,3 +111,29 @@ long stirling2(int n, int k){
     free(curr);
     return output;
 }
+
+
+double stirling2_inexact(long n, long k) {
```

Comment:
I fine with the idea of an approximation but I'd prefer they exist as separate PRs. 
In other words PR 1 contains code in python to use backend stirling2 and PR 2 implements the Temme approximation. 

This is so that the enhancements are-to some extent independent-the PR1 can merge and work fine while PR 2 is in development. 

Also, wouldn't this be 'private' too? e.g. (`_stirling2_inexact`) ?

Reply:
Yeah, agreed. I wanted to add the `exact` keyword arg now even though `exact=False` is not that useful yet, just so we wouldn't have to change the api in the future. I'll follow up with a Temme PR once we get your PR merged into main.

Reply:
Oh, I see what you were saying. I thought you were talking about a Python function, but I had made that private already. In C, leading underscores are reserved for functions and types used by the compiler and standard library. The convention is different than in Python. See [here](https://stackoverflow.com/questions/39625352/why-do-some-functions-in-c-have-an-underscore-prefix)

Reply:
> Yeah, agreed. I wanted to add the `exact` keyword arg now even though `exact=False` is not that useful yet, just so we wouldn't have to change the api in the future. I'll follow up with a Temme PR once we get your PR merged into main.
> 
> And yes, that should be private. I hacked this together hastily this morning.

See my detailed comment at bottom of thread. TL;DR I don't think we should add `exact` argument until we have 2 distinct values (`exact=True` and `exact=False`) that have working codepaths. We can do the `exact=True` as the default codepath ***without*** the `exact` argument at all.